### PR TITLE
Add integration tests module with contract, concurrency, and E2E tests (KOJAK-20)

### PR DESCRIPTION
**Pull request by endrju19**
## Summary

- New `okapi-integration-tests` module (non-published) with comprehensive test suite
- Refactored existing E2E tests from BehaviorSpec to FunSpec (fixing fragile isolation)
- Refactored `PostgresOutboxStore.claimPending()` from raw SQL to type-safe Exposed DSL with `ForUpdateOption.SKIP_LOCKED`
- Added MySQL composite index + `FORCE INDEX` hint to fix InnoDB locking behavior with `SKIP LOCKED`

### Test support infrastructure
- `PostgresTestSupport`, `MysqlTestSupport`, `KafkaTestSupport` — Testcontainers wrappers
- `RecordingMessageDeliverer` — thread-safe test spy with `assertNoAmplification()`

### Store contract tests (16 tests)
Shared via FunSpec extension function, run on both Postgres and MySQL:
- persist/read roundtrip, ordering, limit, status filtering, update, purge, count, empty edge case

### Concurrency tests (4 tests)
- **Deterministic** (CountDownLatch + virtual threads): Thread A holds lock → Thread B gets disjoint entries via SKIP LOCKED
- **Realistic** (CyclicBarrier + virtual thread executor): 5 concurrent processors, verify zero delivery amplification

### Kafka transport integration (5 tests)
- Real Kafka broker via Testcontainers: topic delivery, headers, partition key, null key, success result

### E2E tests (11 tests)
- `HttpEndToEndTest` (Postgres + HTTP): 6 tests including **transaction rollback** and **retry exhaustion** (new scenarios)
- `MysqlHttpEndToEndTest` (MySQL + HTTP): 4 tests
- `KafkaEndToEndTest` (Postgres + Kafka): full pipeline

### Cleanup
- Removed old `OutboxEndToEndTest` and `MysqlOutboxEndToEndTest` (BehaviorSpec, fragile isolation)
- Cleaned up test dependencies from `okapi-spring-boot` and `okapi-mysql`

## Key discovery: MySQL FORCE INDEX

Concurrency tests revealed that without `FORCE INDEX(idx_outbox_status_created_at)`, InnoDB locks ALL examined rows during `SELECT ... FOR UPDATE SKIP LOCKED` (full table scan), not just the LIMIT'd ones. Added composite index migration + FORCE INDEX hint.

## Test plan

- [ ] `./gradlew test -x :okapi-integration-tests:test` — unit tests pass
- [ ] `./gradlew :okapi-integration-tests:test` — all 36 integration tests pass (requires Docker)
- [ ] `./gradlew ktlintCheck` — formatting clean

### CODE REVIEW

File: okapi-mysql/src/main/resources/com/softwaremill/okapi/db/mysql/003__add_claim_index.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+--liquibase formatted sql
+--changeset outbox:002
```
Comment (ramafasa) [inline okapi-mysql/src/main/resources/com/softwaremill/okapi/db/mysql/003__add_claim_index.sql]: should be 003?

File: okapi-postgres/src/main/resources/com/softwaremill/okapi/db/004__add_claim_index.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+--liquibase formatted sql
+--changeset outbox:003
```
Comment (ramafasa) [inline okapi-postgres/src/main/resources/com/softwaremill/okapi/db/004__add_claim_index.sql]: should be 004?

**Review by ramafasa — CHANGES_REQUESTED** [review]
